packages/base/README.md

```diff
@@ -1,2 +1,94 @@
+
+```
+{
+  ...,
+  "appium": {
+    "pluginName": "your-plugin-name",
+    "mainClass": "YourPlugin",
+  },
+  ...
+}
+```
+
+Here the `pluginName` is the name used to activate your plugin by users who have installed it. So if your `pluginName` is `cool-plugin`, they would activate it as follows:
+
+```
+appium --plugins=cool-plugin
+```
+
+The same goes for updating and removing the plugin using the CLI.
+
+The `mainClass` value is the name that Appium should import from your project's entrypoint. So for example if somewhere in your project you have defined your plugin as follows:
+
+```
+class CoolPlugin extends BasePlugin {
+  ...
+}
+```
+
+Then you need to make sure that Appium can import/require this name from your node package:
+
+```
+const { CoolPlugin } = require('someones-cool-plugin')
+```
+
+### 2. Create your Plugin class by extending BasePlugin
+
+As implied above, your project should export a class which extends BasePlugin. This means you need to add BasePlugin as a project dependency:
+
+```
+npm install --save @appium/base-plugin
+```
+
+Then you can create a class which extends it:
+
+```
+import BasePlugin from '@appium/base-plugin'
+
+class CoolPlugin extends BasePlugin {}
+```
+
+### 3. Implement your Plugin class
+
+Your plugin can do one or more of 3 basic things:
+
+1. Update the Appium server object before it starts listening for requests
+2. Handle Appium commands in lieu of the driver which would normally handle them
+3. Add new routes/commands to Appium (which your plugin must then handle)
+
+#### Updating the server
+
+You probably don't normally need to update the Appium server object (which is an Express server having already been configured in a variety of ways). But, for example, you could add new Express middleware to the server to support your plugin's requirements. To update the server you must do two things:
+
+1. Set the class variable `updatesServer` to `true`, so that Appium knows your plugin needs to update the server before launch.
+2. Implement the `async updateServer` method in your class. This method takes two parameters, `expressApp` and `httpServer`, which are the Express app object and the Node HTTP server object respectively. You can do whatever you want with them inside the `updateServer` method.
+
+### Handling Appium commands
+
+This is the most normal behavior for Appium plugins -- to modify or replace the execution of one or more commands. There are two things you must do to override the default command handling:
+
+1. Set the `commands` class variable to either `true` (meaning all commands will be handled by this plugin), or to an array of command names, signifying that your plugin should only handle those particular command names. Curious what command names there are? They are defined in the Appium base driver's [routes.js](https://github.com/appium/appium-base-driver/blob/master/lib/protocol/routes.js) file.
```

Comment:
If someone else wants to do it... :-) I think it would be great for base plugin to export type definitions, or even be written in typescript. This repo might be a good place to think about porting to typescript. But main Appium, not sure.